embed-code-go/test/embedding_instruction_test/embedding_instruction_test.go

```diff
@@ -97,48 +97,39 @@ func xmlAttribute(name string, value string) string {
 }
 
 func TestFalseXML(t *testing.T) {
-	defer func() {
-		if r := recover(); r == nil {
-			t.Errorf("FromXML had to raise a panic, but it didn't.")
-		}
-	}()
-
 	xmlString := "<file=\"org/example/Hello.java\" fragment=\"Hello class\"/>"
 	config := buildConfigWithPreparedFragments()
 
-	embedding_instruction.FromXML(xmlString, config)
+	_, err := embedding_instruction.FromXML(xmlString, config)
+	if err == nil {
+		t.Errorf("FromXML had to raise a panic, but it didn't.")
```

Comment:
Can we please describe the unexpected-expected behaviour in terms of testing our features, not in terms of some calls.

I assume we wanted to test whether a mis-formed XML triggered a parsing error. In this case, let's write our error message in this spirit.